From 0453e672d8c27543d07df246bfcc79439631b419 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 14 Apr 2021 12:12:29 +0300
Subject: [PATCH] [SW-0] greco: remove goya2 residue files

Since goya2 to renamed to greco, this patch removes residue folder
and files of goya2.

Change-Id: I1da2cc4798549ba2006c93e37feba6e4a46ba5c7
---

diff --git a/goya2/preboot.lds.S b/goya2/preboot.lds.S
deleted file mode 100644
index 0e4134f..0000000
--- a/goya2/preboot.lds.S
+++ /dev/null
@@ -1,38 +0,0 @@
-/*
- * (C) Copyright 2017 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
-#include "common.h"
-
-MEMORY { .sram : ORIGIN = PREBOOT_TEXT_BASE, LENGTH = PREBOOT_SIZE }
-
-OUTPUT_FORMAT("elf64-littleaarch64", "elf64-littleaarch64", "elf64-littleaarch64")
-OUTPUT_ARCH(aarch64)
-ENTRY(_start)
-SECTIONS
-{
-	.text : {
-		. = ALIGN(8);
-		preboot/crt0_preboot.o (.text*)
-		* (.text*)
-	} >.sram
-
-	.bss : {
-		. = ALIGN(8);
-		__bss_start = . ;
-		*(.bss*)
-		__bss_end = . ;
-	} > .sram
-
-	.stack : {
-		. = ALIGN(8);
-		__stack = . ;
-		. = . + 4096 ;
-		__stack_top = . ;
-	} > .sram
-
-	.note.gnu.build-id : { *(.note.gnu.build-id) } > .sram
-}
diff --git a/goya2_arc_gnu/security.c b/goya2_arc_gnu/security.c
deleted file mode 100644
index ccb0f84..0000000
--- a/goya2_arc_gnu/security.c
+++ /dev/null
@@ -1,102 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0+
-/*
- * Copyright 2020 HabanaLabs, Ltd.
- * All Rights Reserved.
- *
- */
-
-#include <bitops.h>
-#include <preboot.h>
-#include <string.h>
-#include <uart.h>
-#include <sha256.h>
-#include <goya2/goya2_security.h>
-
-#define EFUSE_SYSCFG_ENABLE_VALIDATION_MASK	BIT(10)
-
-static u32 sp_sram_readl(u32 offset)
-{
-	u32 reg = *((volatile u32 *) (HL_SCRATCHPAD_SRAM_BASE + offset));
-	return reg;
-}
-
-bool preboot_security_is_enabled(void)
-{
-	u32 config_efuse;
-
-	config_efuse = sp_sram_readl(SECURITY_EFUSE_SYS_CFG_OFFSET);
-	return (config_efuse & EFUSE_SYSCFG_ENABLE_VALIDATION_MASK);
-}
-
-bool preboot_security_key_is_revoked(int key_idx)
-{
-	if (key_idx < 0 || key_idx >= SECURITY_KEY_HASH_COUNT)
-		return true;
-
-	key_revoke = sp_sram_readl(SECURITY_EFUSE_KEYS_REVOKE_OFFSET);
-	return !!(key_revoke & BIT(key_idx));
-}
-
-bool preboot_security_key_is_valid(int key_idx, const void *pub_key,
-			 int pub_key_len, const void *pub_key_ext,
-			 int pub_key_ext_len)
-{
-	u32 key_revoke, key_base, hash_calculated, hash_stored;
-	u8 bytes_per_dword = sizeof(u32);
-	u8 value[SHA256_SUM_LEN];
-	u8 key[KEY_SIZE];
-	u8 i;
-
-	if (key_idx < 0 || key_idx >= SECURITY_KEY_HASH_COUNT)
-		return false;
-
-	if (pub_key_len != (KEY_SIZE - KEY_EXT_SIZE)) {
-		preboot_puts_hex("Invalid pub key length: ", pub_key_len, 0);
-		return false;
-	}
-
-	if (pub_key_ext_len != KEY_EXT_SIZE) {
-		preboot_puts_hex("Invalid pub key ext length: ",
-				pub_key_ext_len, 0);
-		return false;
-	}
-
-	/* Concanate public key and public key extension */
-	memcpy(key, pub_key, pub_key_len);
-	memcpy((key + pub_key_len), pub_key_ext, pub_key_ext_len);
-
-	sha256_csum(key, pub_key_len + pub_key_ext_len, value);
-
-	key_base = (key_idx * bytes_per_dword * SECURITY_DWORD_PER_KEY)
-			+ SECURITY_EFUSE_KEYS_OFFSET;
-
-	for (i = 0; i < SECURITY_DWORD_PER_KEY; i++) {
-		memcpy(&hash_calculated, value + (i * bytes_per_dword),
-			bytes_per_dword);
-		hash_stored = sp_sram_readl(key_base + (i * bytes_per_dword));
-		if (hash_calculated != hash_stored)
-			return false;
-	}
-
-	return true;
-}
-
-bool preboot_security_svn_is_valid(int sec_ver)
-{
-	u8 sec_fw_ver, sec_valid_fw_ver;
-	u32 valid_fw_ver;
-
-	sec_fw_ver = SECURITY_FW_VERSION_TO_SECURITY_BYTE(sec_ver);
-	valid_fw_ver = sp_sram_readl(SECURITY_EFUSE_FW_VERSION_OFFSET);
-	sec_valid_fw_ver = SECURITY_FW_VERSION_TO_SECURITY_BYTE(valid_fw_ver);
-
-	if (sec_fw_ver < sec_valid_fw_ver)
-		preboot_puts_hex("Security version too low: ", sec_ver, 0);
-
-	return sec_fw_ver >= sec_valid_fw_ver;
-}
-
-bool preboot_security_key_montgomery_is_valid(void)
-{
-	return true;
-}
